src/adapter/http/src/middleware/authentication_layer.rs

```diff
@@ -53,34 +59,32 @@ impl<Svc> AuthenticationMiddleware<Svc> {
         maybe_access_token: Option<AccessToken>,
     ) -> Result<CurrentAccountSubject, Response> {
         if let Some(access_token) = maybe_access_token {
-            let authentication_service = base_catalog
-                .get_one::<dyn auth::AuthenticationService>()
-                .unwrap();
+            let authentication_service =
+                base_catalog.get_one::<dyn AuthenticationService>().unwrap();
 
             // TODO: Getting the full account info here is expensive while all we need is
             // the caller identity
             match authentication_service
                 .account_info_by_token(access_token.token)
                 .await
             {
-                Ok(account_info) => Ok(CurrentAccountSubject::logged(
-                    account_info.account_name,
-                    account_info.is_admin,
+                Ok(account) => Ok(CurrentAccountSubject::logged(
```

Comment:
There are 2 consumers:
 - HTTP middleware
 - UI (when refreshing page and resolving current account).
 
 UI requires a full account model from token, so we are not winning anything.
 HTTP needs a little less: id, name and "is_admin" marker.
Even if we separate these 2 consumers to different APIs, HTTP needs more data then just ID, so a database trip is required, unless all the data is included into the token. I'm okay with including ID and "is_admin" into the token, but I don't think including account's name into token is now a good idea, as we will not be able to support renaming of accounts - this would invalidate the token.

Reply:
Summary of Slack discussion:
- Kamu token was already changed to carry only account ID
- We agreed NOT to add additional information into the token like account name or `is_admin`
- New token structure breaks Jupyter demo login flows, to fix it we agreed to:
  - Change `kamu system generate-token` command to only accept account ID
    - This command will NOT be used any more in demo login flow
  - Extend `kamu login` to allow authenticating with first-party provider token (passing GH access token)
    - Will require to extend REST endpoint for auth
  - Simplify demo deployment to NOT propagate `JWT_SECRET` to user pods

Reply:
Implemented